x/axelarnet/types/evm_interpreter.go

```diff
@@ -0,0 +1,111 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+
+	"github.com/axelarnetwork/utils/funcs"
+)
+
+var (
+	stringType  = funcs.Must(abi.NewType("string", "string", nil))
+	uint256Type = funcs.Must(abi.NewType("uint256", "uint256", nil))
+	bytesType   = funcs.Must(abi.NewType("bytes", "bytes", nil))
+
+	// abi encoded bytes, with the following format:
+	// wasm method name, argument number, comma seperated argument types, encoded bytes contain actual arguments name and value
+	payloadArguments = abi.Arguments{{Type: stringType}, {Type: uint256Type}, {Type: stringType}, {Type: bytesType}}
+)
+
+type contract struct {
+	// Contract is the address of the wasm contract
+	Contract string `json:"contract"`
+	// Msg is a json struct {"methodName": {"arg1": "val1", "arg2": "val2"}}
+	Msg map[string]interface{} `json:"msg"`
+}
+
+// wasm is the json get passed to the IBC memo field
+type wasm struct {
+	Wasm contract `json:"wasm"`
+}
```

Comment:
Why do we need this `wasm` struct that contains `contract`?

Reply:
it works as a helper wrapper with wasm key, the memo field takes a nested json struct looks like 👇  
```
{
  wasm: {
    {
      msg : {
         "address": address
         "sample_method": {
            "arg1": val1,
            ....
         }
      }
    }
  }
}
```

Reply:
Why does the `wasm` key matter so much?

Reply:
the dest chain middleware checks if the memo contains `wasm` key to decide if it should decode the payload and forward to wasm module